apps/userdashboard/api.py

```diff
@@ -62,17 +58,25 @@ def get_permission_object(self):
         return self.project
 
     def get_queryset(self):
-        all_comments_project = helpers.get_all_comments_project(self.project)
-        num_reports = Count("reports", distinct=True)
-        return all_comments_project.annotate(num_reports=num_reports).annotate(
-            has_reports=ExpressionWrapper(
-                Q(num_reports__gt=0), output_field=BooleanField()
-            )
+        comments = helpers.get_all_comments_project(self.project)
+        comments = comments.annotate(
+            num_reports=Count("reports", distinct=True)
+            + Count("ai_report", distinct=True),
         )
 
+        reported_by = self.request.query_params.get("reported_by")
```

Comment:
more of a personal preference/discussion: instead of doing it here we could also do it in the `ModerationCommentFilterSet`, then the filtering is happening in one place.

It's slightly more verbose though:
```
  @property
    def qs(self):
        queryset = super().qs
        reported_by = self.request.query_params.get("reported_by")
        
        if reported_by == "ai":
            queryset = queryset.filter(ai_report__isnull=False)
        elif reported_by == "users":
            queryset = queryset.filter(reports__isnull=False)
        return queryset
```
What do you think? I'm also fine with how it works now

Reply:
Makes sense. Done.